src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -333,30 +333,45 @@ class FreezeBase : public StackObj {
 protected:
   JavaThread* const _thread;
   ContinuationWrapper& _cont;
-  CONT_JFR_ONLY(FreezeThawJfrInfo _jfr_info;)
   bool _barriers;
   const bool _preempt; // used only on the slow path
+  const intptr_t * const _frame_sp;
 
-  intptr_t *_bottom_address;
+  intptr_t* _bottom_address;
 
   int _size; // total size of all frames plus metadata in words.
   int _align_size;
 
+  intptr_t* _cont_stack_top;
+  intptr_t* _cont_stack_bottom;
+  int _cont_size;
```

Comment:
I know these are in logical grouping and it's a StackObj, but can you move the two bools to after cont_size to minimize alignment gaps.  Also there are three sizes now in this class.  Can we have bigger comments for what each means in the declaration?  I think this will be helpful for the abstraction.

Reply:
Why do you care about alignment? These are like local variables.

Reply:
I removed "int _cont_size"

Reply:
I always care about zeros in memory.

Reply:
You shouldn't :)